### PR TITLE
fix: PK fix for conformance_pack_rule_compliances

### PR DESCRIPTION
**Pull request by spangenberg**
The PK was set on the parent but should be on default `cq_id`.
Also the filter for the Details was not applied.

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Fetch Diagnostics for provider aws:

Resource: config.conformance_packs Type: Database Severity: Error
	Summary: failed to insert to table "aws_config_conformance_pack_rule_compliances": ERROR: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk" (SQLSTATE 23505)
	Detail: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk". [Repeated:36]

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 1
```

With fix:

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 0
cloudquery $
```

### CODE REVIEW

**Review by roneli — CHANGES_REQUESTED** [review]
There seems to be migration errors, based on passed experience would also verify after test fixes.
Fetch previous version -> upgrade to this version with data. 
Downgrade back to previous version. 
Run on clean db

**Comment by spangenberg** [comment]
> There seems to be migration errors, based on passed experience would also verify after test fixes. Fetch previous version -> upgrade to this version with data. Downgrade back to previous version. Run on clean db

It's fixed now, tested everything as well by hand with existing data.

**Review by disq — APPROVED** [review]

File: resources/provider/migrations/timescale/28_v0.11.2.down.sql [hunk 1]
```diff
@@ -1,3 +1,11 @@
 DROP TABLE IF EXISTS aws_elasticbeanstalk_application_versions;
 
 ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
+
+DROP INDEX "aws_config_conformance_pack_rule_compli_cq_fetch_date_cq_id_idx";
```
Comment (roneli) [inline resources/provider/migrations/timescale/28_v0.11.2.down.sql:5]: DROP IF EXISTS

File: resources/provider/migrations/timescale/28_v0.11.2.down.sql [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+
+ALTER TABLE "aws_config_conformance_pack_rule_compliances"
+  DROP CONSTRAINT aws_config_conformance_pack_rule_compliances_pk,
```
Comment (roneli) [inline resources/provider/migrations/timescale/28_v0.11.2.down.sql:8]: IF EXISTS

File: resources/provider/migrations/timescale/28_v0.11.2.down.sql (hunk 1 above)
Comment (roneli) [reply resources/provider/migrations/timescale/28_v0.11.2.down.sql:5]: same with normal

**Review by roneli — CHANGES_REQUESTED** [review]

**Review by roneli — APPROVED** [review]
LGTM

File: resources/provider/migrations/postgres/28_v0.11.2.down.sql [hunk 3]
```diff
@@ -1,3 +1,7 @@
 DROP TABLE IF EXISTS aws_elasticbeanstalk_application_versions;
 
-ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
\ No newline at end of file
+ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
+
+ALTER TABLE IF EXISTS "aws_config_conformance_pack_rule_compliances"
+  DROP CONSTRAINT aws_config_conformance_pack_rule_compliances_pk,
+  ADD CONSTRAINT aws_config_conformance_pack_rule_compliances_pk PRIMARY KEY (conformance_pack_cq_id);
```
Comment (bbernays) [inline resources/provider/migrations/postgres/28_v0.11.2.down.sql:7]: The version needs to be updated to 0.11.3

**Pull request merged**